src/explore-education-statistics-common/src/modules/charts/components/MapBlockInternal.tsx

```diff
@@ -726,6 +503,81 @@ export const MapBlockInternal = ({
       </div>
     </>
   );
-};
+}
+
+function generateFeaturesAndDataGroups({
+  classification,
+  dataSetCategories,
+  dataGroups: groups,
+  selectedDataSetConfig,
+}: {
+  classification: DataClassification;
+  dataSetCategories: MapDataSetCategory[];
+  dataGroups: number;
+  selectedDataSetConfig: DataSetCategoryConfig;
+}): {
+  features: MapFeatureCollection;
+  dataGroups: LegendDataGroup[];
+} {
+  const selectedDataSetKey = selectedDataSetConfig.dataKey;
+  const {
+    unit,
+    decimalPlaces: dataSetDecimalPlaces,
```

Comment:
Really minor, but don't you really need to rename this to dataSetDecimalPlaces now.